nuxeo-retention-web/elements/nuxeo-hold-toggle-button.js

```diff
@@ -145,8 +145,11 @@ class RetentionHoldToggleButton extends mixinBehaviors([FiltersBehavior, FormatB
   }
 
   _hold() {
-    const { isFlexibleRecord } = this.document;
-    const response = isFlexibleRecord ? window.confirm(this.i18n('retention.holdToggleButton.confirm.hold')) : true;
+    const { isFlexibleRecord, isUnderRetentionOrLegalHold } = this.document;
+    const response =
+      isFlexibleRecord && isUnderRetentionOrLegalHold
```

Comment:
Why do we need to check in addition `isUnderRetentionOrLegalHold` before warning?

Reply:
maybe you meant `isFlexibleRecord && isRecord` ?

Reply:
when we unattached the rule on the doocument , `isUnderRetentionOrLegalHold` is the only property which changes from true to false , so in one scenario where a user unattach a rule and then apply legal hold , the pop up confirmation should not appear , so `isUnderRetentionOrLegalHold` is the only property which helps to achieve this.

Reply:
so be it